From 96644b8c00d5cd96859b843d0b11c13e6af64c1b Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:20:05 +0200
Subject: [PATCH] cache: Relax chain_cache_dump filter application

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 18cd6edf86139621dc6bfe0f17a391e46218c027

commit 18cd6edf86139621dc6bfe0f17a391e46218c027
Author: Phil Sutter <phil@nwl.cc>
Date:   Fri Mar 6 17:02:17 2026 +0100

    cache: Relax chain_cache_dump filter application

    While populating chain cache, a filter was only effective if it limited
    fetching to both a table and a chain. Make it apply to 'list chains'
    command as well which at most specifies a family and table.

    Since the code is OK with filter->list fields being NULL, merely check
    for filter to be non-NULL (which is the case if nft_cache_update() is
    called by nft_cmd_enoent_chain()).

    Fixes: 17297d1acbbf ("cache: Filter chain list on kernel side")
    Signed-off-by: Phil Sutter <phil@nwl.cc>
    Reviewed-by: Pablo Neira Ayuso <pablo@netfilter.org>
    Tested-by: Eric Garver <eric@garver.life>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/cache.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cache.c b/src/cache.c
index b82d0fd..5b6f46f 100644
--- a/src/cache.c
+++ b/src/cache.c
@@ -631,7 +631,7 @@ chain_cache_dump(struct netlink_ctx *ctx,
 	const char *chain = NULL;
 	int family = NFPROTO_UNSPEC;
 
-	if (filter && filter->list.table && filter->list.chain) {
+	if (filter) {
 		family = filter->list.family;
 		table = filter->list.table;
 		chain = filter->list.chain;
